snooty/postprocess.py

```diff
@@ -1658,6 +1659,7 @@ def find_toctree_nodes(
         node: Dict[str, Any],
         toc_landing_pages: List[str],
         visited_file_ids: Set[FileId] = set(),
+        external_nodes: List[object] = [],
```

Comment:
Setting default parameters like this tends to be a bit of a code smell, since only one list is ever instantiated:
```
def foo(a=[]):
...     a.append(1)
...     return a
... 
>>> foo()
[1]
>>> foo()
[1, 1]
>>> foo()
[1, 1, 1]
```

If that's *intended* behavior, we should probably document it :D

Reply:
Also see below about typing stuff

Reply:
I think this just shouldn't be a default parameter since the postprocessor can be run many times in one process (e.g. in the vscode extension).